### PR TITLE
Misc 4

### PR DESCRIPTION
**Pull request by jschueller**
- #2563 
- Add DistributionFactory::GetByName(string) method (useful for persalys)
- Enhanced timeout support (add TimeoutException), also new INTERRUPTION enum for optim
- Fixed static library cmake exported target (xref https://gitlab.kitware.com/vtk/vtk/-/merge_requests/10941)
- Fixed ComposedDistribution/Copula alias classes ctors (and restore corresponding headers)
- I noticed that OT is available in vcpkg: https://vcpkg.link/ports/openturns, adding that to the doc

### CODE REVIEW

File: lib/src/Uncertainty/Model/DistributionImplementation.cxx [hunk 1]
```diff
@@ -2357,16 +2357,24 @@ Scalar DistributionImplementation::computeScalarQuantile(const Scalar prob,
...
-  const Scalar root = solver.solve(f, q, leftTau, rightTau, leftCDF, rightCDF);
+  Scalar root = solver.solve(f, p, leftTau, rightTau, leftCDF, rightCDF);
   LOGDEBUG(OSS() << "root=" << root);
+
+  // special case non strictly increasing CDF: retain the inf of the interval veryfing F(x)=p
+  if (computeCDF(root - quantileEpsilon_) >= prob)
+  {
+    solver.setResidualError(0.5 * cdfEpsilon_);
+    root = solver.solve(f, p - cdfEpsilon_, leftTau, root, leftCDF, prob);
+    LOGDEBUG(OSS() << "inf root=" << root);
```
Comment (regislebrun) [inline lib/src/Uncertainty/Model/DistributionImplementation.cxx:2376]: At this point you can add a 1st order correction. You solved $F(x_p)=p-\epsilon$, and you are looking for $\eta$ such that $F(x_p+\eta)=p$. But, up to the first order, $F(x_p+\eta)=F(x_p)+\eta F'(x_p)$, so $p=p-\epsilon+\eta F'(x_p)$, resulting into $\eta=\epsilon/F'(x_p)$. It works only if $F'(x_p)\neq 0$, which is the case for a continuous mixture but not for a discrete distribution or a mix of discrete and continuous distributions.

Comment (jschueller) [reply lib/src/Uncertainty/Model/DistributionImplementation.cxx]: lets continue this in #2564

File: python/src/OptimizationAlgorithmImplementation_doc.i.in [hunk 2]
```diff
@@ -393,7 +393,7 @@ OT_OptimizationAlgorithm_getCheckStatus_doc
 // ---------------------------------------------------------------------
 
 %define OT_OptimizationAlgorithm_setMaximumTimeDuration_doc
-"Accessor to check status flag.
+"Accessor to maximum duration.
```
Comment (josephmure) [inline python/src/OptimizationAlgorithmImplementation_doc.i.in:396]:
```suggestion
"Accessor to the maximum duration.
```

File: python/src/OptimizationAlgorithmImplementation_doc.i.in [hunk 3]
```diff
@@ -406,7 +406,7 @@ OT_OptimizationAlgorithm_setMaximumTimeDuration_doc
 // ---------------------------------------------------------------------
 
 %define OT_OptimizationAlgorithm_getMaximumTimeDuration_doc
-"Accessor to check status flag.
+"Accessor to maximum duration.
```
Comment (josephmure) [inline python/src/OptimizationAlgorithmImplementation_doc.i.in:409]:
```suggestion
"Accessor to the maximum duration.
```

File: python/src/SimulationAlgorithm_doc.i.in [hunk 4]
```diff
@@ -191,7 +191,7 @@ storage_strategy : :class:`~openturns.HistoryStrategy`
 // ---------------------------------------------------------------------
 
 %feature("docstring") OT::SimulationAlgorithm::setMaximumTimeDuration
-"Accessor to check status flag.
+"Accessor to maximum duration.
```
Comment (josephmure) [inline python/src/SimulationAlgorithm_doc.i.in:194]:
```suggestion
"Accessor to the maximum duration.
```

File: python/src/SimulationAlgorithm_doc.i.in [hunk 5]
```diff
@@ -201,7 +201,7 @@ maximumTimeDuration : float
 // ---------------------------------------------------------------------
 
 %feature("docstring") OT::SimulationAlgorithm::getMaximumTimeDuration
-"Accessor to check status flag.
+"Accessor to maximum duration.
```
Comment (josephmure) [inline python/src/SimulationAlgorithm_doc.i.in:204]:
```suggestion
"Accessor to the maximum duration.
```

**Review by josephmure — APPROVED** [review]
Small change and then good to go IMO.

**Comment by jschueller** [comment]
fyi @m-balesdent for the optim/simulation changes

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**